src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts

```diff
@@ -84,9 +92,15 @@ export class GLASSDataSubmissionDefaultRepository implements GLASSDataSubmission
         const objects =
             (await this.globalStorageClient.getObject<GLASSDataSubmissionItem[]>(namespace))?.filter(object => {
                 const amrModule = modules.find(module => module.name === "AMR")?.id;
+                const amrIndividualModule = modules.find(module => module.name === "AMR - Individual")?.id;
+                const egaspModule = modules.find(module => module.name === "EGASP")?.id;
 
-                return object.module === amrModule;
+                if (module === "AMR") return object.module === amrModule;
+                else if (module === "AMRIndividual") return object.module === amrIndividualModule;
+                else if (module === "EGASP") return object.module === egaspModule;
+                else return [];
```

Comment:
Be careful here because filter always returns truthy or falsy values, so here `[]` would be evaluated as true. So if none of the above cases is met, always returns a truthy value and therefore no element is filtered in `objects`